packages/flutter_test/test/accessibility_test.dart

```diff
@@ -0,0 +1,177 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  group('text contrast guideline', () {
+    testWidgets('black text on white background - Text Widget - direct style', (WidgetTester tester) async {
+      final SemanticsHandle handle = tester.ensureSemantics();
+      await tester.pumpWidget(_boilerplate(
+        const Text(
+          'this is a test',
+          style: TextStyle(fontSize: 14.0, color: Colors.black),
+        ),
+      ));
+      await expectLater(tester, meetsGuideline(textContrastGuideline));
```

Comment:
Can you add a test that ensures that a failing matcher actually produces the expected errors?

Reply:
Same issue as above

Reply:
Made the guideline classes (but not constructors) public so they can be unit tested, add failure message test cases.